### PR TITLE
[6.2.z] Implemented Entity __eq__ method comparing all fields (#350)

### PR DESCRIPTION
**Pull request by renzon**
(cherry picked from commit ecf91be)

### CODE REVIEW

**Comment by renzon** [comment]
Cherry picking to be able to use on z streams branches. Worth mentioning that there is no 6.1.z branch on nailgun which would refer to 6.1.z satellite z stream.

File: nailgun/entity_mixins.py [hunk 1]
```diff
@@ -191,6 +196,15 @@ def _payload(fields, values):
                 values[field_name + '_ids'] = [
                     entity.id for entity in values.pop(field_name)
                 ]
+            elif isinstance(field, ListField):
+                def parse(obj):
```
Comment (rochacbruno) [inline nailgun/entity_mixins.py]: Coveralls says this inner function is not covered, can you write a simple test case that covers this elif, or if does not make sense to test include a `... ListField): # pragma: no cover` comment here

File: nailgun/entity_mixins.py [hunk 2]
```diff
@@ -1304,3 +1374,26 @@ def search_filter(entities, filters):
...
+
+    :param obj: entity or any json serializable object
+
+    :return: serializable object
+
+    """
+    if isinstance(obj, Entity):
+        return obj.to_json_dict()
+
+    if isinstance(obj, dict):
```
Comment (rochacbruno) [inline nailgun/entity_mixins.py]: All this cases below are not covered by tests, can you include simple test cases for each of those data types?

File: nailgun/entity_mixins.py [hunk 3]
```diff
@@ -510,6 +523,63 @@ def __repr__(self):
...
+        ...         'id': 1,
+        ...         'name': 'Nailgun Org',
+        ...     }
+        >>> org = entities.Organization(config.ServerConfig('foo'), \*\*kwargs)
+        >>> org.to_json()
+        '{"id": 1, "name": "Nailgun Org"}'
+
+        :return: str
+        """
+        return std_json.dumps(self.to_json_dict())
```
Comment (rochacbruno) [inline nailgun/entity_mixins.py]: This return is also not being covered by test shoult it?

File: nailgun/entity_mixins.py (hunk 2 above)
Comment (renzon) [inline nailgun/entity_mixins.py]: Strange, it seems cherry pick not worked as I expected. Going to check the test file because it was covered.

**Comment by renzon** [comment]
@rochacbruno fixed test coverage.

**Comment by rochacbruno** [comment]
Coveralls is still mad, but looks like a configuration problem, or some areas we miss a `pragma: no cover` comment

**Pull request merged**